### PR TITLE
ci: add maven cache

### PR DESCRIPTION
**Pull request by bxb100**
close: gh-73

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -40,9 +40,22 @@ jobs:
...
+    runs-on: ${{ matrix.os }}
     steps:
       - uses: actions/checkout@v3
+      - name: Cache local Maven repository
+        uses: actions/cache@v3
+        timeout-minutes: 5
+        with:
+          path: |
+            ~/.m2/repository/*/*/*
+            !~/.m2/repository/org/apache/pulsar
```
Comment (tisonkun) [inline .github/workflows/ci.yml:55]: This project is not `pulsar`, lol.

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -63,6 +76,16 @@ jobs:
     runs-on: ${{ matrix.os }}
     steps:
       - uses: actions/checkout@v3
+      - name: Cache local Maven repository
+        uses: actions/cache@v3
+        timeout-minutes: 5
+        with:
+          path: |
+            ~/.m2/repository/*/*/*
+            !~/.m2/repository/org/apache/pulsar
```
Comment (tisonkun) [inline .github/workflows/ci.yml:85]: ditto

You may take a look at https://github.com/actions/cache for use case.

**Review by tisonkun — CHANGES_REQUESTED** [review]
I think...either we just don't set the path parameter, or exclude hawkeye path instead.

File: .github/workflows/ci.yml (hunk 1 above)
Comment (bxb100) [reply .github/workflows/ci.yml:55]: awk, missed checking this

**Comment by bxb100** [comment]
fixed, @tisonkun

**Review by tisonkun — APPROVED** [review]
Thanks for your quick follow-up @bxb100!

No sorry 😆 You help to solve an issue that benefit this project and you should be proud of it.

Merging...

I rerun the workflow and the macos job consume less time from 1m45s to 15s. IIUC it can work better in schedule scenario.

**Comment by bxb100** [comment]
Thanks

**Pull request merged**